lib/web/mage/gallery/gallery.js

```diff
@@ -339,29 +342,37 @@ define([
                             }
 
                             if (isTouchEnabled) {
-                                settings.breakpoints[pair[0]].options.arrows = false;
+                                settings.breakpoints[mediaQuery].options.arrows = false;
 
-                                if (settings.breakpoints[pair[0]].options.fullscreen) {
-                                    settings.breakpoints[pair[0]].options.fullscreen.arrows = false;
+                                if (settings.breakpoints[mediaQuery].options.fullscreen) {
+                                    settings.breakpoints[mediaQuery].options.fullscreen.arrows = false;
                                 }
                             }
 
-                            settings.api.updateOptions(settings.breakpoints[pair[0]].options, true);
-                            $.extend(true, config, settings.breakpoints[pair[0]]);
-                            settings.activeBreakpoint = settings.breakpoints[pair[0]];
+                            settings.api.updateOptions(settings.breakpoints[mediaQuery].options, true);
+                            $.extend(true, config, settings.breakpoints[mediaQuery]);
+                            settings.activeBreakpoint = settings.breakpoints[mediaQuery];
+
+                            if (!isInitialized[mediaQuery]) {
```

Comment:
I would recommend removing this check (condition). Simply `isInitialized[mediaQuery] = true;` will be more sufficient from the performance and code readability perspective 
```suggestion

```

Reply:
Thanks for review. Change pushed.